terraform/secrets.tf

```diff
@@ -0,0 +1,287 @@
+  secret = google_secret_manager_secret.database_user.name
+}
+
+resource "google_secret_manager_secret_iam_member" "database_user" {
+  secret_id  = google_secret_manager_secret.database_user.id
+  role       = "roles/secretmanager.secretAccessor"
+  member     = "serviceAccount:${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+  depends_on = [google_secret_manager_secret.database_user]
+}
+
+resource "google_secret_manager_secret" "database_password" {
+  secret_id = "database_password"
+  replication {
+    automatic = true
+  }
+}
+
+data "google_secret_manager_secret_version" "database_password" {
+  secret = google_secret_manager_secret.database_password.name
+}
+
+resource "google_secret_manager_secret_iam_member" "database_password" {
+  secret_id  = google_secret_manager_secret.database_password.id
+  role       = "roles/secretmanager.secretAccessor"
+  member     = "serviceAccount:${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+  depends_on = [google_secret_manager_secret.database_password]
+}
+
+resource "google_secret_manager_secret" "database_name" {
+  secret_id = "database_name"
+  replication {
+    automatic = true
+  }
+}
+
+data "google_secret_manager_secret_version" "database_mame" {
+  secret = google_secret_manager_secret.database_name.name
+}
+
+resource "google_secret_manager_secret_iam_member" "database_name" {
+  secret_id  = google_secret_manager_secret.database_name.id
+  role       = "roles/secretmanager.secretAccessor"
+  member     = "serviceAccount:${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+  depends_on = [google_secret_manager_secret.database_name]
+}
+
+resource "google_secret_manager_secret" "database_socket" {
+  secret_id = "database_socket"
+  replication {
+    automatic = true
+  }
+}
+
+data "google_secret_manager_secret_version" "database_socket" {
+  secret = google_secret_manager_secret.database_socket.name
+}
+
+resource "google_secret_manager_secret_iam_member" "database_socket" {
+  secret_id  = google_secret_manager_secret.database_socket.id
+  role       = "roles/secretmanager.secretAccessor"
+  member     = "serviceAccount:${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+  depends_on = [google_secret_manager_secret.database_socket]
+}
+
+resource "google_secret_manager_secret" "slack_team_id" {
+  secret_id = "slack_team_id"
+  replication {
+    automatic = true
+  }
+}
+
+data "google_secret_manager_secret_version" "slack_team_id" {
+  secret = google_secret_manager_secret.slack_team_id.name
+}
+
+resource "google_secret_manager_secret_iam_member" "slack_team_id" {
+  secret_id  = google_secret_manager_secret.slack_team_id.id
+  role       = "roles/secretmanager.secretAccessor"
+  member     = "serviceAccount:${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+  depends_on = [google_secret_manager_secret.slack_team_id]
```

Comment:
Similarly for these you shouldn't need to explicitly set up `depends_on` since you're referencing the resource in the `secret_id`